Feat/change api endpoint

mathild3r opened the pull request:
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [x] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

_Explain why this PR exists_

replace the /issues endpoint by /aggregated-issues

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_
Some fields are missing in the new endpoint (upgradePath and package manager for example)

### More information

- [Jira ticket SC-0000](https://snyksec.atlassian.net/browse/SC-0000)
- [Link to documentation](https://github.com/snyk/snyk-delta/wiki/)

### Screenshots

_Visuals that may help the reviewer_

aarlaud commented on src/lib/index.ts:
```diff
@@ -96,21 +96,21 @@ const getDelta = async(snykTestOutput = '', debugMode = false, setPassIfNoBaseli
     } else {
       snykProject = await snyk.getProjectIssues(baselineOrg,baselineProject)
+
+      console.log(JSON.stringify(snykProject))
```
are we sure to leave this trace here?

aarlaud replied:
probably should get dropped

aarlaud commented on src/lib/snyk/displayOutput.ts:
```diff
@@ -33,6 +33,8 @@ const displayNewVulns = (
       console.log('Security Vulnerabilities:');
     }
+
+    console.log(newVulns)
```
should probably drop this, or put it behind debug function

aarlaud commented on src/lib/utils/issuesUtils.ts:
```diff
@@ -23,6 +41,80 @@ const isVulnerablePathNew = (monitoredSnapshotPathArray: Array<string>, currentS
+            const {cvssScore, ...everythingElse} = aggregatedIssueData
+            issuesPostResponse.ok = true
+
+            const abc: IssueDataWithMissingField = { from : [], package: '', upgradePath: [], version: '', isPatched: false, isIgnored: false, cvssScore: parseInt(cvssScore), ...everythingElse }  
```
Let's rename this here to something more meaningful

aarlaud commented on test/fixtures/apiResponses/Untitled-1.json:
```diff
@@ -0,0 +1,102 @@
+{
```
Give this file a better filename than untitled. hard to know what this fixture reflects otherwise.

aarlaud commented on test/fixtures/apiResponses/Untitled-2.json:
```diff
@@ -0,0 +1,54 @@
+{
```
same thing, meaningful filename please.

aarlaud commented on test/fixtures/apiResponses/java-goof.json:
```diff
@@ -2429,4 +2429,6 @@
     "packageManager": "maven"
-  }
\ No newline at end of file
+}
```
I'd say remove those empty lines

aarlaud commented on test/fixtures/apiResponses/test-gomod-aggregated.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "issues": []
```
that doesn't look right, does it?

aarlaud commented on test/lib/snyk/issues.test.ts:
```diff
@@ -333,7 +333,7 @@ describe('Test issues functions', () => {
       const newVulns: Array<any> = [];
       displayNewVulns(newVulns, 'inline');
-      expect(consoleOutput).toEqual([]);
+      expect(consoleOutput).toEqual([[]]);
```
why are we expecting a double nested array here?

aarlaud commented on src/lib/snyk/snyk.ts:
```diff
@@ -59,4 +70,45 @@ const getProjectDepGraph = async (orgID: string, projectID: string) => {
+}
+
+const getUpgradePath = async (orgID: string, projectID: string, issueId: string) => {
+  const projectIssuePaths = await new snykClient.Org({ orgId: orgID })
```
We're not pulling all the pages here, are we? We need to pull all the paths

aarlaud replied:
check, you might have a getAll method available that does that for you already.

mathild3r commented:
I changed the function from getAll to a loop that gets one result per page and loop if there is next page as I couldn't test or be sure that getAll will get all the pages (sounds unlikely that it won't but couldn't be sure) I updated the tests and fixtures and tested manually on a goof project.

aarlaud commented:
:tada: This PR is included in version 1.5.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/snyk-delta/v/1.5.0)
- [GitHub release](https://github.com/snyk-tech-services/snyk-delta/releases/tag/v1.5.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
